### ISSUE TITLE
Add comments to code based on feedback

### ISSUE THREAD

**Issue by gliechtenstein** [intercellular/cell#137]
This library has an unconventional architecture. And the code initially had near 0 annotation. This made it difficult to understand what's going on.

So I have started adding comments https://github.com/intercellular/cell/blob/develop/cell.js 

I would love to improve it, so please let me know if there are parts that are unclear and I'll update or add more comments.

**Comment by piranna** [comment]
Could you stick them to 80 columns so they are easier to read? :-)

**Comment by gliechtenstein** [comment]
@piranna I had actually started updating the code to reflect your feedback but the code became too long, and the functions became too far from one another separated by long paragraphs of comments. 

I think each option has pros and cons. Too many lines of code makes it hard to get a quick birds eye view of the code. Also it's good to have comments but having too many lines of code gives the impression that the code is more complex than it actually is (humans are irrational beings).

I'm going to just stash the changes I've made so far, but if many people want this option I'll think about it. Thanks for the feedback!
